### ISSUE TITLE
Zoom (not unzoom) performance on iOS Safari is not good

### ISSUE THREAD

**Issue by rpearce** [rpearce/react-medium-image-zoom#227]
Related to https://github.com/rpearce/react-medium-image-zoom/issues/222#issuecomment-593684993

Example website: https://sunknudsen.com/privacy-guides/brute-forcing-your-very-own-vanity-onion-address-at-11646mhs

**Comment by sunknudsen** [comment]
Updated link: https://sunknudsen.com/privacy-guides/brute-forcing-your-very-own-vanity-onion-address-at-11-646mh-s

**Comment by rpearce** [comment]
Thanks, @sunknudsen. I need to get this to be smoother overall and fixup iOS.

**Comment by rpearce** [comment]
FYI I'm almost done completely rewriting this library into:
1. a vanilla JS package that is ~3kb (minified & gzipped)
1. a react hook that uses 👆  and is maybe 100-150 bytes 

This and all the other issues should be closed out by it. I'm _almost_ there (working out some API stuff).

Thanks for your patience

**Comment by sunknudsen** [comment]
Hey @rpearce! Didn’t you just rewrite the whole thing in TypeScript? Looking forward to your refactor!

**Comment by rpearce** [comment]
@sunknudsen It's still in TypeScript, but I'll be releasing this as two packages (so far):
* `@rpearce/image-zoom`, which I'll have an alpha out today or tomorrow for, is vanilla JavaScript (written in TypeScript) that can be included on any webpage, regardless of framework. It weighs in at ~2.9KB gzipped and minifed.
* `react-medium-image-zoom` version 5 is looking like it'll be simply a react hook that calls the appropriate functions for `@rpearce/image-zoom`. This'll be just a few hundred bytes of code, if that. Example usage:
```js
const { ref } = useImageZoom({ /* image zoom options here */ })

return <img ... ref={ref} />
```

One more thing: both of these will work on any element you provide and should be very accessible.

**Comment by rpearce** [comment]
There is an alpha release out (no documentation nor tests yet): [5.0.0-alpha.1](https://www.npmjs.com/package/react-medium-image-zoom/v/5.0.0-alpha.1)

This has condensed the "react" nature of this to nothing more than a react hook wrapping https://www.npmjs.com/package/@rpearce/image-zoom/v/0.1.0-alpha.2.

For gatsby, it looks like their `gatsby-image` might not accept `ref` (let me know if it does), which sucks, but good news: you can do this:
```js
const { ref } = useImageZoom({ /* image zoom options here */ })

return (
  <div ref={ref}>
    <Image ... />
  </div>
```
However, it is not guaranteed to have all the refined parts that using it directly on an `img` tag would.

I'm considering doing a check to see if the current target is NOT an `img`, but the child IS an `img`, then just attach it to that -- maybe through a `type` flag with the value `'find-first-img'` or something, but I haven't thought it out fully.

**Comment by rpearce** [comment]
This will be resolved soon and can be beta previewed:

```sh
npm i --save react-medium-image-zoom@5.0.0-beta2.6
```

The trick with this and having crisp images when going from small to large sizes on mobile is to start out by specifying that the zoom image is the size of the window (max X or Y axes) and scale the image _down_ to the size of the original.

This way, when it does the "zoom", it's going from like `scale(0.23)` to `scale(1)`.

**Comment by rpearce** [comment]
Should be closed now with the release of `5.0.0`
